iphone/Classes/TiUITabGroup.m

```diff
@@ -636,6 +636,10 @@ - (void)open:(id)args
 
 - (void)close:(id)args
 {
+  if ([self.proxy _hasListeners:@"close"]) {
+    [self.proxy fireEvent:@"close" withObject:event];
+  }
```

Comment:
It should be called event-driven (by delegates that actually execute the closing. Let me find a better place to put it.

Reply:
Isn't this `close` method called by [windowDidClose](https://github.com/garymathews/titanium_mobile/blob/88fb153fb72d4d0fb8a560800c62e3394e92737a/iphone/Classes/TiUITabGroupProxy.m#L157) anyway?

Reply:
It is (by `windowWillClose`), but if you call it manually, the `close` event may fire before the window actually finishing closing. I'll prepare a test-case tomorrow.

Reply:
Ok, so let's do this (as the above test-case does not actually close the tab group):
```js
var win = Ti.UI.createWindow(),
	tabGroup = Ti.UI.createTabGroup();
	tab = Ti.UI.createTab({
		title: 'Tab',
		window: win
	});
  
win.addEventListener('open', function() {
  setTimeout(function() {
    tabGroup.close();
  }, 2000);
})

tabGroup.addEventListener('close', function() {
	console.log('TabGroup.close()');
});

tabGroup.addTab(tab);
tabGroup.open();
```
It will wait 2 seconds after the application finished launching and close the tab-group. First, `windowWillClose` is called to cleanup the navigation stack. Once done, it will will invoke it's super-class that will clean up all child views (TiViewProxy instances). 

Then, finally, the TiWindowProxy, which the TiUITabGroupProxy inherits from cleans up (dismisses the actual controller, fire the child close events) and sends `windowDidClose` to `TiUITabGroupProxy`.

So after all, the close event in its current state would fire before the tabgroup actually finished closing, which is different from the behavior of other root view controllers (like window or split-window on iOS).

And as you stated above, because the class inherits from TiWindowProxy, is should fire the event already (but the link you referenced was linking to the wrong method).